packages/manager/modules/billing/src/autoRenew/actions/terminate-all-dom/controller.js

```diff
@@ -0,0 +1,129 @@
+import set from 'lodash/set';
+import kebabCase from 'lodash/kebabCase';
+import map from 'lodash/map';
+import filter from 'lodash/filter';
+import { BillingService } from '@ovh-ux/manager-models';
+
+const DOMAIN_SERVICE_TYPE = 'DOMAIN';
+export default class {
+  /* @ngInject */
+  constructor($translate, $http, $q, atInternet, BillingAutoRenew) {
+    this.$translate = $translate;
+    this.$http = $http;
+    this.$q = $q;
+    this.atInternet = atInternet;
+    this.BillingAutoRenew = BillingAutoRenew;
+  }
+
+  $onInit() {
+    this.isLoading = false;
+    this.cancelAllDom = true;
+  }
+
+  selectAllDomains() {
+    this.domains.forEach((domain) => set(domain, 'selected', true));
+  }
+
+  cancelServices() {
+    const domainsSelected = filter(this.domains, (domain) => domain.selected);
+    if (this.cancelAllDom || domainsSelected.length > 0) {
```

Comment:
The method `cancelServices` should be accessible only if you have `domainSelected.length > 0` (Using `primary-disabled` (See: https://ovh.github.io/ovh-ui-kit/?path=/docs/design-system-components-modal-webcomponent--simple), cause it is weird to allow main action, which will do the second action (cancelling modal). This way you could simplify a bit this method (`if` 3 times in a row)